Port the remaining WebCore/DOM types to the new serialization format

gavin-apple opened the pull request:
#### cbe22e46962fff4854bb3128d3eb6a3e0b9cba76
<pre>
Port the remaining WebCore/DOM types to the new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=251584">https://bugs.webkit.org/show_bug.cgi?id=251584</a>
&lt;rdar://problem/104959020&gt;

Reviewed by Alex Christensen.

This change includes moving the following types:
    - EventInit
    - MessageWithMessagePorts
    - StorageAccessWasGranted
    - StorageAccessPromptWasShown
    - StorageAccessScope
    - StorageAccessQuickResult
    - RequestStorageAccessResult
    - Exception
    - ExceptionData
    - MessagePortIdentifier
    - SecurityPolicyViolationEventInit
    - CSPViolationReportBody

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/DocumentStorageAccess.h:
(WebCore::RequestStorageAccessResult::encode const): Deleted.
(WebCore::RequestStorageAccessResult::decode): Deleted.
* Source/WebCore/dom/EventInit.h:
(WebCore::EventInit::encode const): Deleted.
(WebCore::EventInit::decode): Deleted.
* Source/WebCore/dom/Exception.h:
(WebCore::Exception::encode const): Deleted.
(WebCore::Exception::decode): Deleted.
* Source/WebCore/dom/ExceptionData.h:
(WebCore::ExceptionData::encode const): Deleted.
(WebCore::ExceptionData::decode): Deleted.
* Source/WebCore/dom/MessageChannel.cpp:
(WebCore::generateMessagePorts):
* Source/WebCore/dom/MessagePortIdentifier.h:
(): Deleted.
(WebCore::MessagePortIdentifier::encode const): Deleted.
(WebCore::MessagePortIdentifier::decode): Deleted.
* Source/WebCore/dom/PortIdentifier.h: Copied from Source/WebCore/dom/EventInit.h.
* Source/WebCore/dom/SecurityPolicyViolationEvent.cpp:
(WebCore::SecurityPolicyViolationEventInit::SecurityPolicyViolationEventInit):
* Source/WebCore/dom/SecurityPolicyViolationEvent.h:
(WebCore::SecurityPolicyViolationEventInit::SecurityPolicyViolationEventInit): Deleted.
(WebCore::SecurityPolicyViolationEventInit::encode const): Deleted.
(WebCore::SecurityPolicyViolationEventInit::decode): Deleted.
* Source/WebCore/dom/messageports/MessageWithMessagePorts.h:
(WebCore::MessageWithMessagePorts::encode const): Deleted.
(WebCore::MessageWithMessagePorts::decode): Deleted.
* Source/WebCore/page/csp/CSPViolationReportBody.cpp:
(WebCore::CSPViolationReportBody::CSPViolationReportBody):
(WebCore::CSPViolationReportBody::create):
* Source/WebCore/page/csp/CSPViolationReportBody.h:
(WebCore::CSPViolationReportBody::encode const): Deleted.
(WebCore::CSPViolationReportBody::decode): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isMessagePortAlive const):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::SerializedScriptValue&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::SerializedScriptValue&gt;&gt;::decode):
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::ReportBody&gt;&gt;::encode):
(IPC::ArgumentCoder&lt;RefPtr&lt;WebCore::ReportBody&gt;&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/259893@main">https://commits.webkit.org/259893@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4441bc86ae1cbeea9a82385467124ed0a266aadd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106322 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15376 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115516 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175619 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110233 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16853 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6592 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98539 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115205 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112086 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12802 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95783 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40358 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/109796 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94677 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27426 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82048 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8617 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28778 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9126 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5348 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14757 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48324 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6836 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10691 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/Headers.cmake:
```diff
@@ -786,6 +786,7 @@ set(WebCore_PRIVATE_FRAMEWORK_HEADERS
     dom/OverflowEvent.h
     dom/PointerEvent.h
     dom/PointerEventTypeNames.h
+    dom/PortIdentifier.h
```
I think this should be called "MessagePortIdentifier" to make it more clear what type of port it is.
Test failures look related.

gavin-apple replied:
Test failure was unrelated - it was the SerializedScriptValue issue mentioned.

```
struct MessagePortIdentifier {
    ProcessIdentifier processIdentifier;
    PortIdentifier portIdentifier;

#if !LOG_DISABLED
    String logString() const;
#endif
};
```

MessagePortIdentifier is already the name of the struct that wraps this. It's used all over the codebase. TBH, I think MessagePortIdentifier should probably become ProcessQualified - but this particular type has some additional HashTraits applied to it; so that might not be a trivial change.

achristensen07 reviewed (approved):
All the good names are already taken

gavin-apple committed:
Port the remaining WebCore/DOM types to the new serialization format
https://bugs.webkit.org/show_bug.cgi?id=251584
<rdar://problem/104959020>

Reviewed by Alex Christensen.

This change includes moving the following types:
    - EventInit
    - MessageWithMessagePorts
    - StorageAccessWasGranted
    - StorageAccessPromptWasShown
    - StorageAccessScope
    - StorageAccessQuickResult
    - RequestStorageAccessResult
    - Exception
    - ExceptionData
    - MessagePortIdentifier
    - SecurityPolicyViolationEventInit
    - CSPViolationReportBody

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/DocumentStorageAccess.h:
(WebCore::RequestStorageAccessResult::encode const): Deleted.
(WebCore::RequestStorageAccessResult::decode): Deleted.
* Source/WebCore/dom/EventInit.h:
(WebCore::EventInit::encode const): Deleted.
(WebCore::EventInit::decode): Deleted.
* Source/WebCore/dom/Exception.h:
(WebCore::Exception::encode const): Deleted.
(WebCore::Exception::decode): Deleted.
* Source/WebCore/dom/ExceptionData.h:
(WebCore::ExceptionData::encode const): Deleted.
(WebCore::ExceptionData::decode): Deleted.
* Source/WebCore/dom/MessageChannel.cpp:
(WebCore::generateMessagePorts):
* Source/WebCore/dom/MessagePortIdentifier.h:
(): Deleted.
(WebCore::MessagePortIdentifier::encode const): Deleted.
(WebCore::MessagePortIdentifier::decode): Deleted.
* Source/WebCore/dom/PortIdentifier.h: Copied from Source/WebCore/dom/EventInit.h.
* Source/WebCore/dom/SecurityPolicyViolationEvent.cpp:
(WebCore::SecurityPolicyViolationEventInit::SecurityPolicyViolationEventInit):
* Source/WebCore/dom/SecurityPolicyViolationEvent.h:
(WebCore::SecurityPolicyViolationEventInit::SecurityPolicyViolationEventInit): Deleted.
(WebCore::SecurityPolicyViolationEventInit::encode const): Deleted.
(WebCore::SecurityPolicyViolationEventInit::decode): Deleted.
* Source/WebCore/dom/messageports/MessageWithMessagePorts.h:
(WebCore::MessageWithMessagePorts::encode const): Deleted.
(WebCore::MessageWithMessagePorts::decode): Deleted.
* Source/WebCore/page/csp/CSPViolationReportBody.cpp:
(WebCore::CSPViolationReportBody::CSPViolationReportBody):
(WebCore::CSPViolationReportBody::create):
* Source/WebCore/page/csp/CSPViolationReportBody.h:
(WebCore::CSPViolationReportBody::encode const): Deleted.
(WebCore::CSPViolationReportBody::decode): Deleted.
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isMessagePortAlive const):
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
* Source/WebKit/Scripts/webkit/tests/MessageArgumentDescriptions.cpp:
(IPC::serializedIdentifiers):
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<RefPtr<WebCore::SerializedScriptValue>>::encode):
(IPC::ArgumentCoder<RefPtr<WebCore::SerializedScriptValue>>::decode):
(IPC::ArgumentCoder<RefPtr<WebCore::ReportBody>>::encode):
(IPC::ArgumentCoder<RefPtr<WebCore::ReportBody>>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/259893@main
